sympy/solvers/ode.py

```diff
@@ -3968,6 +3980,168 @@ def _frobenius(n, m, p0, q0, p, q, x0, x, c, check=None):
 
     return frobdict
 
+def _nth_algebraic_match(eq, func):
+    r"""
+    Matches any differential equation that nth_algebraic can solve. Uses
+    `sympy.solve` but teaches it how to integrate derivatives.
+
+    This involves calling `sympy.solve` and does most of the work of finding a
+    solution (apart from evaluating the integrals).
+    """
+
+    # Each integration should generate a different constant
+    constants = iter(numbered_symbols(prefix='C', cls=Symbol, start=1))
+    constant = lambda: next(constants, None)
+
+    # Like Derivative but "invertible"
+    class diffx(Function):
+        def inverse(self):
+            # We mustn't use integrate here because fx has been replaced by _t
+            # in the equation so integrals will not be correct while solve is
+            # still working.
+            return lambda expr: Integral(expr, var) + constant()
+
+    # Replace derivatives wrt the independent variable with diffx
+    def replace(eq, var):
+        def expand_diffx(*args):
+            differand, diffs = args[0], args[1:]
+            toreplace = differand
+            for v, n in diffs:
+                for _ in range(n):
+                    if v == var:
+                        toreplace = diffx(toreplace)
+                    else:
+                        toreplace = Derivative(toreplace, v)
+            return toreplace
+        return eq.replace(Derivative, expand_diffx)
+
+    # Restore derivatives in solution afterwards
+    def unreplace(eq, var):
+        return eq.replace(diffx, lambda e: Derivative(e, var))
+
+    # The independent variable
+    var = func.args[0]
+    subs_eqn = replace(eq, var)
+    try:
+        solns = solve(subs_eqn, func)
+    except NotImplementedError:
+        solns = []
+
+    solns = [unreplace(soln, var) for soln in solns]
+    solns = [Equality(func, soln) for soln in solns]
+    return {'var':var, 'solutions':solns}
+
+def ode_nth_algebraic(eq, func, order, match):
```

Comment:
This should be added to `doc/src/modules/solvers/ode.rst`.

Reply:
Done